CRM/Core/BAO/Address.php

```diff
@@ -1320,7 +1320,7 @@ public static function buildOptions($fieldName, $context = NULL, $props = array(
             $props['country_id'] = $config->defaultContactCountry;
           }
         }
-        if (!empty($props['country_id']) && $context !== 'validate') {
+        if (!empty($props['country_id'])) {
```

Comment:
So this is the line I think is risky - the single field handling below is a bit hacky but mitigated by the unit test IMHO so I'm OK with it 

I found where this was added - @colemanw was fixing a webform bug

https://issues.civicrm.org/jira/browse/CRM-15215